core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -859,14 +861,14 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
    * `ControllerContext.partitionsBeingReassigned` must be populated with all partitions being reassigned before this
    * method is invoked to avoid premature deletion of the `reassign_partitions` znode.
    */
-  private def removePartitionFromReassignedPartitions(topicPartition: TopicPartition) {
-    controllerContext.partitionsBeingReassigned.get(topicPartition).foreach { reassignContext =>
+  private def removePartitionsFromReassignedPartitions(partitionsToBeRemoved: Set[TopicPartition]) {
+    controllerContext.partitionsBeingReassigned.filterKeys(partitionsToBeRemoved).values.foreach { reassignContext =>
```

Comment:
Would it be better to do this the other way around? That is, iterate over `partitionsToBeRemoved` (which should be smaller) and get the context from `partitionsBeingReassigned`.

Reply:
@ijuma Good point. Thanks for the review. I have updated the patch as suggested and rebased it onto latest trunk.